cip/CIP2015-10-12-CREATE.adoc

```diff
@@ -0,0 +1,136 @@
+----
+
+[source, cypher]
+.Create multiple nodes with properties from a parameter list of maps.
+----
+UNWIND { mapList } AS map
+CREATE (n:Node)
+SET n = map
+----
+
+[source, cypher]
+// It seems the . operator doesn't allow multiple lines, and we want two sentences here.
+.Create two nodes with properties from two parameter maps. Parameter keys and values will be the same as those of the map.
+----
+CREATE (:Cat {catProps})
+CREATE (:Dog {dogProps})
+----
+
+[source, cypher]
+.Create three nodes, connect them with two relationships and return the whole pattern as a path.
+----
+CREATE (andres { name: 'Andres' })
+CREATE (neo)
+CREATE (michael { name: 'Michael' })
+CREATE (andres)-[r1:WORKS_AT]->(neo)
+CREATE (michael)-[r2:WORKS_AT]->(neo)
+RETURN (andres)-[r1]->(neo)<-[r2]-(michael) AS p
+----
+
+=== Syntax
+[source, ebnf]
+----
+create       = "CREATE", {pattern}, [{",", pattern}] ;
+pattern      = node-pattern | rel-pattern ;
+node-pattern = "(", [identifier], [{label}], [property], ")" ;
+rel-pattern  = "(", identifier, ")", in-rel | out-rel, "(", identifier, ")" ;
+out-rel      = "-[", [identifier], type, [property], "]->" ;
+in-rel       = "<-[", [identifier], type, [property], "]-" ;
+label        = ":", identifier ;
+type         = label ;
+property     = "{", param | {identifier, ":", literal | param}, "}" ;
+param        = "{", identifier, "}" ;
+literal      = string | number ;
+----
+
+=== Semantics
+
+`CREATE` is used to create new nodes and relationships in the graph.
+Each `CREATE` will create exactly one node or one relationship between two pre-existing nodes per row.
+`CREATE` has no return value, and ending a query with a `CREATE` clause will yield an empty result.
+
+=== Interaction with existing features
+
+`CREATE` on a node pattern will cause the query to fail in the commit phase if the to-be-created node would violate a uniqueness constraint in the database.
+
+This proposal drops some functionality around how we handle creating nodes and relationships:
+
+- `CREATE (a {param})` where `{param}` is a list of maps where one node is created for each item in the list.
+See the example with `UNWIND` above for a way to achieve the same behavior.
+
+- `CREATE (a:L {prop: 42})-[:R]->(b:B)` where both nodes and relationships are created from the same pattern.
+With this proposal one would have to be more specific and write `CREATE (a:L {prop:42}), (b:B), (a)-[:R]->(b)`.
+
+- `CREATE p = (andres { name: 'Andres' })-[:WORKS_AT]->(neo)<-[:WORKS_AT]-(michael { name:'Michael' }) RETURN p`, where a full path is created and returned.
+See the final example above for how to achieve the same behaviour.
+
+We believe that the implicit creation of nodes when specifying a relationship pattern is causing more complexity than whatever benefit it provides to the user, and thus forcing the user to be specific about what to create will benefit both users and implementors of Cypher.
```

Comment:
"more complexity than whatever benefit it provides to the user" -> "more complexity than any benefit provided to the user"